index.bs

```diff
@@ -2694,34 +4614,155 @@ partial interface MLGraphBuilder {
         - *SumSquare*: Compute the sum of the square of all the input values along the axes.
 </div>
 
-### The relu() method ### {#api-mlgraphbuilder-relu}
+<details open>
+  <summary>
+    To <dfn for="MLGraphBuilder" data-lt="reduce-op">create reduce operation</dfn> given |op|, |input| and |options|, run the following steps:
+  </summary>
+  <div algorithm=reduce class=algorithm-steps>
+    1. [=Assert=]: |op| is one of "reduceL1", "reduceL2", "reduceLogSum", "reduceLogSumExp", "reduceMax", "reduceMean", "reduceMin", "reduceProduct", "reduceSum", "reduceSumSquare".
+    1. [=Assert=]: the type of |input| is {{MLOperand}}.
+    1. If |options| is `undefined`, let |options| be a new {{MLReduceOptions}} object with |options|.{{MLReduceOptions/keepDimensions}} set to `false` and |options|.{{MLReduceOptions/axes}} set to `null`.
```

Comment:
The default values of |options|.{{MLReduceOptions/axes}} should be {0, ...., N-1} rather than null.

Reply:
The WebIDL specifies `null`, and the step was removed because the bindings already take care of the initialization. Should this change, please file an issue and let's handle it separately.

Reply:
The description in the spec says:

> The values in the sequence must be in the range [0, N-1] where N is the [=rank=] of the input tensor. 
If not present, all dimensions are reduced.

So in the algorithm, we should check that when `axes` are given, the values are in the correct range.